Sources/CacheAdvance/CacheHeaderHandle.swift

```diff
@@ -155,13 +177,42 @@ final class CacheHeaderHandle {
         try handle.write(data: currentHeader.asData)
     }
 
-    private func resetFile() throws {
-        try handle.truncate(at: 0)
-
-        // Now that we've started from scratch, write a new header.
-        try writeHeaderData()
+    private func memoizedMetadata() throws -> Metadata {
+        if let persistedMetadata = self.persistedMetadata {
+            return persistedMetadata
+        } else {
+            return try Metadata(headerData: try readHeaderData())
+        }
```

Comment:
Since we're only memoizing `Metadata` that is static, we're fine.

Of course, another process could smash what is on disk and we'd get out of sync. But... I think that'd be user error.